### PR TITLE
docs: Precise README.md instructions

### PR DESCRIPTION
**Pull request by jjerphan**
This has helped me installing and using `scikit-learn-nv`.

The second commit was needed for my setup, but you might want to discard it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,23 +6,39 @@
...
+
+   ```commandline
+   conda install -c conda-forge -c rapidsai-nightly -c nvidia --no-channel-priority pylibraft=23.04 cupy rmm
+   ```
+
+3. Activate the environment:
+
+   ```commandline
+   conda activate sklearn-nv
+   ```
```
Comment (betatim) [inline README.md]: Can we make this step 2? The current step (2) lost the `-n sklearn-nv`. Overall I think putting the activation before more `conda install` commands is what more people are used to. The fact that we both overlooked the `-n` at first and now lost it from the command IMHO  shows that activating first is less error prone :D

File: README.md [hunk 2]
```diff
@@ -6,23 +6,39 @@
...
+
+4. Checkout and install the development version of scikit-learn from the `feature/engine-api` branch
+   (see [`scikit-learn#25535`](https://github.com/scikit-learn/scikit-learn/pull/25535). Using [`gh`](https://cli.github.com/):
+
+   ```commandline
+   gh repo clone scikit-learn/scikit-learn
+   (cd scikit-learn; gh pr checkout 25535)
+   ```
```
Comment (betatim) [inline README.md:35]:
```
5. Install the scikit-learn development version with `pip install --no-build-isolation -e scikit-learn`
```
adds the command to install scikit-learn back. The suggestion feature of GitHub is somehow broken when I put the ``` in the suggestion :-/

**Comment by betatim** [comment]
Do you understand why `--no-channel-priority` is (sometimes) needed? I'm not an expert on conda and channel ordering, etc. The fact that it works for some but not others makes me wonder if we could for example reorder the `-c` commands to achieve the same? And/or does it depend on the fact that some of us use `conda` and some `mamba` (aliased to `conda`)?

**Comment by jjerphan** [comment]
> Do you understand why --no-channel-priority is (sometimes) needed? I'm not an expert on conda and channel ordering, etc.

Some people set [strict channel priority](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority) to prevent problems. This allows removing the priority for this command.

> The fact that it works for some but not others makes me wonder if we could for example reorder the -c commands to achieve the same?

It might be possible.

Nevertheless, I think it is better if we simply remove the flag (people might know their setup and add it accordingly). This would mean reverting 454ed1b. What do you think? :slightly_smiling_face:

File: README.md (hunk 2 above)
Comment (jjerphan) [reply README.md:35]: Has b36eee7 addressed it?

File: README.md [hunk 3]
```diff
@@ -6,23 +6,46 @@
...
+
+4. Checkout the development version of scikit-learn from the `feature/engine-api` branch
+   (see [`scikit-learn#25535`](https://github.com/scikit-learn/scikit-learn/pull/25535). Using [`gh`](https://cli.github.com/):
+
+   ```commandline
+   gh repo clone scikit-learn/scikit-learn
+   (cd scikit-learn; gh pr checkout 25535)
+   ```
+
+5. Install the scikit-learn development version with:
```
Comment (betatim) [inline README.md]:
```suggestion
5. Install the scikit-learn development version by running the following in the folder you checked the code out to:
```

**Pull request merged**